twenty-first/src/util_types/algebraic_hasher.rs

```diff
@@ -9,19 +9,51 @@ use crate::shared_math::x_field_element::{XFieldElement, EXTENSION_DEGREE};
 
 pub const RATE: usize = 10;
 
+/// The hasher [Domain] differentiates between the modes of hashing.
+///
+/// The main purpose of declaring the domain is to prevent collisions between
+/// different types of hashing by introducing defining differences in the way
+/// the hash function's internal state (e.g. a sponge state's capacity) is
+/// initialized.
 #[derive(Debug, PartialEq, Eq)]
 pub enum Domain {
+    /// The `VariableLength` domain is used for hashing objects that potentially
+    /// serialize to more than one rate of field elements.
     VariableLength,
+
+    /// The `FixedLength` domain is used for hashing objects that always fit
+    /// within one rate of fields elements, e.g. a pair of [Digest].
     FixedLength,
 }
 
 pub trait SpongeHasher: Clone + Send + Sync {
     type SpongeState;
 
-    fn absorb_init(input: &[BFieldElement; RATE]) -> Self::SpongeState;
+    /// Initialize a sponge state without absorbing.
```

Comment:
I don't think the fact that no absorption happens needs to be mentioned explicitly.